### PR TITLE
Assure MySql 8.0.3 compatibility

### PR DESCRIPTION
**Pull request by christianheimann**
Needed quoting 'system' keyword, re-became keyword in 8.0.3
Also quoting other keywords that caused no problems: component, export
https://dev.mysql.com/doc/refman/8.0/en/keywords.html

### CODE REVIEW

File: Services/Component/classes/class.ilCachedComponentData.php [hunk 1]
```diff
@@ -133,7 +133,7 @@ protected function readFromDB() {
 			$parent = $rec['parent'];
 			$this->subobj_for_parent[$parent][] = $rec;
 		}
-		$set = $ilDB->query('SELECT DISTINCT(id) AS sid, parent, il_object_def.* FROM il_object_def, il_object_subobj WHERE NOT (system = 1) AND NOT (sideblock = 1) AND subobj = id');
+		$set = $ilDB->query('SELECT DISTINCT(id) AS sid, parent, il_object_def.* FROM il_object_def, il_object_subobj WHERE NOT (`system` = 1) AND NOT (sideblock = 1) AND subobj = id');
```
Comment (mjansenDatabay) [inline Services/Component/classes/class.ilCachedComponentData.php]: Maybe `\ilDBInterface::quoteIdentifier` should be used instead.

File: setup/classes/class.ilObjDefReader.php [hunk 2]
```diff
@@ -139,9 +139,9 @@ function handlerBeginTag($a_xml_parser,$a_name,$a_attribs)
 					}
 	
 					$this->current_object = $a_attribs["id"];
-					$ilDB->manipulateF("INSERT INTO il_object_def (id, class_name, component,location,".
+					$ilDB->manipulateF("INSERT INTO il_object_def (id, class_name, `component`,location,".
```
Comment (mjansenDatabay) [inline setup/classes/class.ilObjDefReader.php]: Maybe `\ilDBInterface::quoteIdentifier` should be used instead.

File: setup/classes/class.ilObjDefReader.php [hunk 3]
```diff
@@ -139,9 +139,9 @@ function handlerBeginTag($a_xml_parser,$a_name,$a_attribs)
...
 						"checkbox,inherit,translate,devmode,allow_link,allow_copy,rbac,default_pos,".
-						"default_pres_pos,sideblock,grp,system,export,repository,workspace,administration,".
+						"default_pres_pos,sideblock,grp,`system`,`export`,repository,workspace,administration,".
```
Comment (mjansenDatabay) [inline setup/classes/class.ilObjDefReader.php]: Maybe `\ilDBInterface::quoteIdentifier` should be used instead.

**Review by mjansenDatabay — COMMENTED** [review]
Thx for your contribution @christianheimann .

@chfsx I am not 100% sure, but AFAIR developers should use the respective `ilDBInterface` method instead of backticks?

File: setup/classes/class.ilObjDefReader.php (hunk 2 above)
Comment (alex40724) [reply setup/classes/class.ilObjDefReader.php]: Or even better $ilDB->insert. Afaik this does quoteIdentifier automatically.

**Comment by alex40724** [comment]
@christianheimann Thanks for the PR. Could you please file a bug report for the error you are getting in the ILIAS tracker? This might need another solution depending on the commands being involved (I expect there to be more of these issues).

**Comment by christianheimann** [comment]
I created a bug report with [issue nr 0026511](https://mantis.ilias.de/view.php?id=26511), referenced to category «setup» as one error occured there. I'm sorry to deliver not more detail this time.

**Comment by klees** [comment]
Hi everyone,

I had a chat with @chfsx. We decided to close the ticket but still accept the PR in a modified version using `quoteIdentifiers`:

https://mantis.ilias.de/view.php?id=26511

Best regards!

**Comment by christianheimann** [comment]
@klees Thanks for the acceptance and the reply, changed using quoteIdentifiers, so hope this fits now.

**Comment by kergomard** [comment]
@chfsx :  Could you please have a look at this PR. It seems to have gotten stuck at the end of last year. Thank you very much!

**Comment by christianheimann** [comment]
I added a new quotation to a keyword in ilParticipant class; updated branch so can be finally accepted easily.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
@christianheimann Can you please check why two methods in `Services/Certificate` are renamed? Maybe adding these files was not intended.

**Comment by christianheimann** [comment]
@mjansenDatabay Thanks for this input, due to a git issue with renamed files (when only case changes), i had this problem. Reset as leading 5-4 branch, so PR only contains necessary changes due to mysql compatibility.

**Comment by christianheimann** [comment]
@mjansenDatabay Thanks for this input, due to a git issue with renamed files (when only case changes), i had this problem. Reset as leading 5-4 branch, so PR only contains necessary changes due to mysql compatibility.

**Comment by christianheimann** [comment]
@mjansenDatabay Thanks for this input, due to a git issue with renamed files (when only case changes), i had this problem. Reset as leading 5-4 branch, so PR only contains necessary changes due to mysql compatibility.

**Comment by christianheimann** [comment]
@mjansenDatabay Thanks for this input, due to a git issue with renamed files (when only case changes), i had this problem. Reset as leading 5-4 branch, so PR only contains necessary changes due to mysql compatibility.

**Comment by christianheimann** [comment]
@mjansenDatabay Thanks for this input, due to a git issue with renamed files (when only case changes), i had this problem. Reset as leading 5-4 branch, so PR only contains necessary changes due to mysql compatibility.

**Comment by christianheimann** [comment]
@mjansenDatabay Thanks for this input, due to a git issue with renamed files (when only case changes), i had this problem. Reset as leading 5-4 branch, so PR only contains necessary changes due to mysql compatibility.

**Review by mjansenDatabay — APPROVED** [review]

**Review by klees — APPROVED** [review]

**Comment by alex40724** [comment]
Changes are in generally ok for me.

**Review by alex40724 — APPROVED** [review]

**Comment by kergomard** [comment]
@chfsx : Could you have a look at this and merge if ok. It would be a pity if after all the efforts we would run into merge and other errors, because we leave this too long.

**Comment by klees** [comment]
Hi @chfsx,

this came up again in our last TB meeting on the 12-07. Is there anything new regarding this PR? Is there anything we could do to speed up the process?

Best regards!

**Comment by Amstutz** [comment]
Hi @chfsx,

this came up again in our last TB meeting on the 12-01-2021. Is there anything new regarding this PR? Is there anything we could do to speed up the process?

Best regards!

**Comment by Amstutz** [comment]
Hi @chfsx,

this came up again in our last TB meeting on the 23-02-2021. Is there anything new regarding this PR? Is there anything we could do to speed up the process?

Best regards!

**Comment by chfsx** [comment]
Hi @Amstutz 
There's nothing which could speed up that. merging is currently not possible due to conflicts and it would be better to have the PR against trunk. Since we support MySQL 8 from release_6, I'll fix that from trunk back. I may take some of the changes @christianheimann suggested, thanks for the work.
I close the PR and will fix it in trunk, release_7 and release_6. And if its possible I'll cherry-pick it to release_5-4 as well, bt I can't guarantee that.

**Pull request closed without merging**